src/osyris/plot/render.py

```diff
@@ -51,15 +52,23 @@ def render(x=None, y=None, data=None, logx=False, logy=False, ax=None):
             getattr(wrappers, func)(ax, x, y, item["data"], **item["params"]))
 
         need_cbar = False
+
+        if func == "line_integral_convolution":
+            if "color" in item["params"]:
+                need_cbar = True
+
         if func in ["contourf", "pcolormesh"]:
             need_cbar = True
         if (func == "scatter") and ("c" in item["params"]):
             if not isinstance(item["params"]["c"], str):
                 need_cbar = True
         if need_cbar and cbar:
-            cb = plt.colorbar(mpl_objects[-1], ax=ax, cax=None)
-            cb.set_label(make_label(name=item["name"], unit=item["unit"]))
+            if func == "line_integral_convolution":
```

Comment:
I would try to avoid the if statement here by making the `line_integral_convolution` return an image object, instead of the axes.
I can see above that `mpl_objects` uses `.append`, so it is actually accepting only a single object.
If you want to return both images, you could modify `mpl_objects` to use `.extend` instead. You would then have to modify the other wrapper functions to return a list, containing a single object.
But this is probably a useful change for the future anyway, as other wrappers may generate more than one mpl objects.

Reply:
I can avoid the extra if statement and still render the correct colorbar the following way:

`if "cbar" in item["params"]:
            cbar = item["params"]["cbar"]
            del item["params"]["cbar"]
        else:
            cbar = True

        mpl_objects.extend(
            getattr(wrappers, func)(ax, x, y, item["data"], **item["params"]))

        need_cbar = False

        ind_render = -1
        name = item["name"]
        unit = item["unit"]

        if func == "line_integral_convolution" and "color" in item["params"]:
            need_cbar = True
            ind_render = -2
            name = item["params"]["color"].name
            unit = item["params"]["color"].unit.units

        if func in ["contourf", "pcolormesh"]:
            need_cbar = True
        if (func == "scatter") and ("c" in item["params"]):
            if not isinstance(item["params"]["c"], str):
                need_cbar = True
        if need_cbar and cbar:
            cb = plt.colorbar(mpl_objects[ind_render], ax=ax, cax=None)
            cb.set_label(make_label(name=name, unit=unit))
            cb.ax.yaxis.set_label_coords(-1.1, 0.5)`

Since the line_integral_convolution wrapper returns two images, we need to instruct the renderer to only create a colorbar for the first ax.imshow that gets returned (which is the first item in the return list of said wrapper function, hence the ind_render = -2). Let me know if you're okay with this syntax.